<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Statistical Analysis of Microarray data</title>
  <meta name="description" content="Statistical Analysis of Microarray data" />
  <meta name="generator" content="bookdown 0.17.2 and GitBook 2.6.7" />

  <meta property="og:title" content="Statistical Analysis of Microarray data" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Statistical Analysis of Microarray data" />
  
  
  

<meta name="author" content="Based on Gonzalo, Ricardo and Sanchez-Pla, Alex (2019)" />


<meta name="date" content="2020-03-29" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="introduction.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="summaryabstract.html"><a href="summaryabstract.html"><i class="fa fa-check"></i><b>1</b> Summary/Abstract</a><ul>
<li class="chapter" data-level="1.1" data-path="summaryabstract.html"><a href="summaryabstract.html#key-words"><i class="fa fa-check"></i><b>1.1</b> Key Words</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>2</b> Introduction</a></li>
<li class="chapter" data-level="3" data-path="materials.html"><a href="materials.html"><i class="fa fa-check"></i><b>3</b> Materials</a><ul>
<li class="chapter" data-level="3.1" data-path="materials.html"><a href="materials.html#software"><i class="fa fa-check"></i><b>3.1</b> Software</a></li>
<li class="chapter" data-level="3.2" data-path="materials.html"><a href="materials.html#data"><i class="fa fa-check"></i><b>3.2</b> Data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="methods.html"><a href="methods.html"><i class="fa fa-check"></i><b>4</b> Methods</a><ul>
<li class="chapter" data-level="4.1" data-path="methods.html"><a href="methods.html#environment-preparation"><i class="fa fa-check"></i><b>4.1</b> Environment preparation:</a></li>
<li class="chapter" data-level="4.2" data-path="methods.html"><a href="methods.html#prepare-the-data-for-the-analysis"><i class="fa fa-check"></i><b>4.2</b> Prepare the data for the analysis</a></li>
<li class="chapter" data-level="4.3" data-path="methods.html"><a href="methods.html#packages-installation-in-r"><i class="fa fa-check"></i><b>4.3</b> Packages installation in R</a></li>
<li class="chapter" data-level="4.4" data-path="methods.html"><a href="methods.html#read-the-cel-files"><i class="fa fa-check"></i><b>4.4</b> Read the CEL files</a></li>
<li class="chapter" data-level="4.5" data-path="methods.html"><a href="methods.html#quality-control-of-raw-data"><i class="fa fa-check"></i><b>4.5</b> Quality control of raw data</a></li>
<li class="chapter" data-level="4.6" data-path="methods.html"><a href="methods.html#data-normalization"><i class="fa fa-check"></i><b>4.6</b> Data normalization</a></li>
<li class="chapter" data-level="4.7" data-path="methods.html"><a href="methods.html#quality-control-of-normalized-data"><i class="fa fa-check"></i><b>4.7</b> Quality control of normalized data</a></li>
<li class="chapter" data-level="4.8" data-path="methods.html"><a href="methods.html#batch-detection"><i class="fa fa-check"></i><b>4.8</b> Batch detection</a></li>
<li class="chapter" data-level="4.9" data-path="methods.html"><a href="methods.html#detecting-most-variable-genes"><i class="fa fa-check"></i><b>4.9</b> Detecting most variable genes</a></li>
<li class="chapter" data-level="4.10" data-path="methods.html"><a href="methods.html#filtering-least-variable-genes"><i class="fa fa-check"></i><b>4.10</b> Filtering least variable genes</a></li>
<li class="chapter" data-level="4.11" data-path="methods.html"><a href="methods.html#saving-normalized-and-filtered-data"><i class="fa fa-check"></i><b>4.11</b> Saving normalized and filtered data</a></li>
<li class="chapter" data-level="4.12" data-path="methods.html"><a href="methods.html#defining-the-experimental-setup-the-design-matrix"><i class="fa fa-check"></i><b>4.12</b> Defining the experimental setup: The design matrix</a></li>
<li class="chapter" data-level="4.13" data-path="methods.html"><a href="methods.html#defining-comparisons-with-the-contrasts-matrix"><i class="fa fa-check"></i><b>4.13</b> Defining comparisons with the Contrasts Matrix</a></li>
<li class="chapter" data-level="4.14" data-path="methods.html"><a href="methods.html#model-estimation-and-gene-selection"><i class="fa fa-check"></i><b>4.14</b> Model estimation and gene selection</a></li>
<li class="chapter" data-level="4.15" data-path="methods.html"><a href="methods.html#obtaining-lists-of-differentially-expressed-genes"><i class="fa fa-check"></i><b>4.15</b> Obtaining lists of differentially expressed genes</a></li>
<li class="chapter" data-level="4.16" data-path="methods.html"><a href="methods.html#gene-annotation"><i class="fa fa-check"></i><b>4.16</b> Gene Annotation</a></li>
<li class="chapter" data-level="4.17" data-path="methods.html"><a href="methods.html#visualizing-differential-expression"><i class="fa fa-check"></i><b>4.17</b> Visualizing differential expression</a></li>
<li class="chapter" data-level="4.18" data-path="methods.html"><a href="methods.html#multiple-comparisons"><i class="fa fa-check"></i><b>4.18</b> Multiple Comparisons</a></li>
<li class="chapter" data-level="4.19" data-path="methods.html"><a href="methods.html#heatmaps"><i class="fa fa-check"></i><b>4.19</b> Heatmaps</a></li>
<li class="chapter" data-level="4.20" data-path="methods.html"><a href="methods.html#biological-significance-of-results"><i class="fa fa-check"></i><b>4.20</b> Biological Significance of results</a></li>
<li class="chapter" data-level="4.21" data-path="methods.html"><a href="methods.html#summary-of-results"><i class="fa fa-check"></i><b>4.21</b> Summary of results</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="notes.html"><a href="notes.html"><i class="fa fa-check"></i><b>5</b> Notes</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Statistical Analysis of Microarray data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Statistical Analysis of Microarray data</h1>
<p class="author"><em>Based on Gonzalo, Ricardo and Sanchez-Pla, Alex (2019)</em></p>
<p class="date"><em>March 29, 2020</em></p>
</div>
<div id="summaryabstract" class="section level1">
<h1><span class="header-section-number">1</span> Summary/Abstract</h1>
<p>Microarray data analysis has been one of the most important hits in the interaction between statistics and bioinformatics in the last two decades. The analysis of microarray data can be done in different ways using different tools. In this chapter a typical workflow for analyzing microarray data using R and Bioconductor packages is presented. The workflow starts with the raw data -binary files obtained from the hybridization process- and goes through a series of steps: Reading raw data, Quality Check, Normalization, Filtering, Selection of differentially expressed genes, Comparison of selected lists and Analysis of Biological Significance. The implementation of each step in R is described through a use case that goes from raw data until the analysis of biological significance. Data and code for the analysis are provided in a github repository<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>.</p>
<div id="key-words" class="section level2">
<h2><span class="header-section-number">1.1</span> Key Words</h2>
<p>Microarrays, Bioconductor, R, Differential Expression</p>
</div>
</div>
<div class="footnotes">
<hr />
<ol start="1">
<li id="fn1"><p><a href="https://github.com/ASPteaching/Omics_Data_Analysis-Case_Study_1-Microarrays" class="uri">https://github.com/ASPteaching/Omics_Data_Analysis-Case_Study_1-Microarrays</a><a href="summaryabstract.html#fnref1" class="footnote-back">↩</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="introduction.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
},
"search": false
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
